src/Styleguide/Components/Lightbox/Lightbox.tsx

```diff
@@ -0,0 +1,286 @@
+          onTouchMove={this.detectActivity}
+          innerRef={this.state.deepZoomRef as any}
+        >
+          <Box
+            position="absolute"
+            top={space(3) / 2}
+            right={space(3) / 2}
+            zIndex={1001}
+          >
+            <CloseButton onClick={() => this.hide()} />
+          </Box>
+          <Flex
+            position="absolute"
+            width="100%"
+            justifyContent="center"
+            zIndex={1001}
+            bottom={space(2)}
+          >
+            <FadeTransition
+              in={this.state.showZoomSlider}
+              timeout={{ enter: 50, exit: 150 }}
+            >
+              <Slider
+                min={slider.min}
+                max={slider.max}
+                step={slider.step}
+                value={slider.value}
+                onChange={this.onSliderChanged}
+                onZoomInClicked={() => this.zoomIn()}
+                onZoomOutClicked={() => this.zoomOut()}
+              />
+            </FadeTransition>
+          </Flex>
+        </DeepZoomContainer>
+      </FadeTransition>
+    )
+  }
+
+  renderPortal = () => {
+    return this.state.element
+      ? ReactDOM.createPortal(this.renderLightbox(), this.state.element)
+      : null
+  }
+
+  render() {
+    const { children } = this.props
+    const modifiedChildren = React.Children.map(children, child =>
+      React.cloneElement(child as any, {
+        style: { cursor: "zoom-in" },
+        onClick: this.show,
+      })
+    )
+    return (
+      <React.Fragment>
+        {this.renderPortal()}
+        {modifiedChildren}
+      </React.Fragment>
+    )
+  }
+
+  postRender = () => {
+    this.state.viewer.addHandler(
+      "zoom",
+      bind(throttle(this.onZoomChanged, 50), this)
+    )
+    this.state.viewer.addHandler(
+      "tile-drawn",
+      once(() => {
+        this.setState({
+          slider: {
+            ...this.state.slider,
+            min: this.state.viewer.viewport.getMinZoom(),
+            max: this.state.viewer.viewport.getMaxZoom(),
+            value: this.state.viewer.viewport.getHomeZoom(),
+          },
+        })
+      })
+    )
+    this.detectActivity()
+  }
+
+  show = event => {
+    this.setState({ shown: true, showZoomSlider: true })
+  }
+
+  hide = () => {
+    this.setState({ shown: false })
+    if (this.state.viewer) {
+      this.state.viewer.destroy()
+      this.state.viewer = null
+    }
+    document.removeEventListener(KEYBOARD_EVENT, this.handleKeyPress)
+    clearTimeout(this.state.activityTimer)
+  }
+
+  handleKeyPress = event => {
+    if (event && event.key === "Escape") {
+      this.hide()
+    }
+  }
+
+  detectActivity = throttle(() => {
+    clearTimeout(this.state.activityTimer)
+    this.setState({
+      showZoomSlider: true,
+      activityTimer: setTimeout(() => {
+        this.setState({
+          showZoomSlider: false,
+        })
+      }, HIDE_ZOOM_SLIDER_AFTER),
+    })
+  }, 500) as () => void
```

Comment:
I got some type issues w/ throttle... I think it's because it wasn't finding an import of the result of its return type... something like that. I don't actually ever change it though, so I think it should say the same.

Reply:
If we’re using [these typings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/49a86971ef16956a22089bd5e8b2aac9650725c0/types/lodash/index.d.ts#L11254-L11258) then I’d expect it to inflect the type of the function, but as it’s generic over `T` you should also be able to do: `throttle<() => void>(() => { … })` instead of `as () => void`.